python/ray/serve/exceptions.py

```diff
@@ -38,3 +39,18 @@ def __str__(self):
             return f"Request {self._request_id} was cancelled."
         else:
             return "Request was cancelled."
+
+
+@PublicAPI(stability="alpha")
+class DeploymentUnavailableError(RayServeException):
+    """Raised when a Serve deployment is unavailable to receive requests.
+
+    This can be because the deployment failed to deploy.
+    """
+
+    def __init__(self, deployment_id: DeploymentID):
+        self._deployment_id = deployment_id
+
+    @property
+    def message(self) -> str:
+        return f"{self._deployment_id} is unavailable"
```

Comment:
nit: this should say "because it failed to deploy" since that's the only reason it can happen today